### PR TITLE
[Deposit Summary] Expanded UI version with animation

### LINKED ISSUE #10069: [Mobile Payments] Implement expanded UI version with animation

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #10069

### Description

The PR adds UI that shows more data about deposits/balances. It also adds animation.

The PR missing:
* Dark theme support
* Tabs per currency
* Proper colors for the deposit statuses
* Deposit Scheduler is just hardcoded for now
* Learn more action

### Testing instructions

On the account with WCPay go to
* More -> Payments
* Notice the deposit info
* Click on it and notice the animation

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4923871/505d7cc4-ec00-425b-bb4e-2cff3b7be3e5

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryScreen.kt [hunk 1]
```diff
@@ -96,10 +143,252 @@ fun PaymentsHubDepositSummaryView(
...
+    backgroundColor: Int,
+    textColor: Int
+) {
+    Box(
+        modifier = Modifier
+            .background(
+                color = colorResource(backgroundColor),
+                shape = RoundedCornerShape(4.dp)
+            )
+            .padding(horizontal = 8.dp, vertical = 4.dp)
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryScreen.kt]: 💡 Let's use dimen resources instead

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryScreen.kt [hunk 2]
```diff
@@ -96,10 +143,252 @@ fun PaymentsHubDepositSummaryView(
...
+
+@Composable
+private fun AdditionInfo(
+    nextDeposit: PaymentsHubDepositSummaryState.Deposit?,
+    lastDeposit: PaymentsHubDepositSummaryState.Deposit?,
+) {
+    Column {
+        Column(
+            modifier = Modifier
+                .padding(start = 16.dp, end = 16.dp, top = 10.dp, bottom = 24.dp)
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryScreen.kt]: 💡 How about dimen resources instead?

**Review by samiuelson — APPROVED** [review]
LGTM

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryScreen.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryScreen.kt]: @samiuelson I replaced them in here 4e257eb235046d5983a908ff6b9d48be6cc1683e

But I am wondering what value do you see in this? I was thinking about potential better support for different screen types, but I guess that won't be on the level of padding, but more like another layout for tablets?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryScreen.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryScreen.kt]: Done here - https://github.com/woocommerce/woocommerce-android/commit/4e257eb235046d5983a908ff6b9d48be6cc1683e

**Pull request merged**